src/areas/render.rs

```diff
@@ -37,10 +38,10 @@ struct Ui {
 }
 
 impl Ui {
-    fn run_ui(cli_args: &cli::Args, areas: Areas) -> Result<(), ErrorString> {
+    fn run_ui(cli_args: &cli::Args, quitter: Quitter, areas: Areas) -> Result<(), ErrorString> {
         let mut ui = Ui::new(cli_args, areas)?;
         ui.run_main_loop()?;
-        ui.quit();
+        quitter.quit();
```

Comment:
Just a heads up: as written, this won't be called if `run_main_loop()` exists with error, or if it unwinds via panic. You probably want to add `impl Drop for Quitter` that calls cleanup, which guarantees it'll be called on unwind / early return because of `?`. Alternatively, there's the [scopeguard](https://docs.rs/scopeguard/) crate that has a handy Go-style `defer!` macro.